agreementbot/agreementworker.go

```diff
@@ -331,15 +331,73 @@ func (b *BaseAgreementWorker) AgreementLockManager() *AgreementLockManager {
 	return b.alm
 }
 
+func (b *BaseAgreementWorker) checkPolicyCompatibility(workerId string, wi *InitiateAgreement, nodePolicy *policy.Policy, businessPolicy *policy.Policy, mergedServicePolicy *externalpolicy.ExternalPolicy, nodeArch string, msgPrinter *message.Printer) (bool, *policy.Policy, error) {
+
+        if compatible, reason, _, consumPol, err := compcheck.CheckPolicyCompatiblility(nodePolicy, businessPolicy, mergedServicePolicy, "", msgPrinter); err != nil {
+                glog.Warning(BAWlogstring(workerId, fmt.Sprintf("error checking policy compatibility. %v.", err.Error())))
+                return false, nil, err
+        } else {
+                if compatible {
+                        if glog.V(5) {glog.Infof(BAWlogstring(workerId, fmt.Sprintf("Node %v is compatible", wi.Device.Id)))}
+                        return true, consumPol, nil
+                } else {
+                        glog.Warningf(BAWlogstring(workerId, fmt.Sprintf("failed matching node policy %v and %v, error: %v", wi.ProducerPolicy, wi.ConsumerPolicy, reason)))
+                        return false, nil, nil
+                }
+        }
+}
+
 func (b *BaseAgreementWorker) InitiateNewAgreement(cph ConsumerProtocolHandler, wi *InitiateAgreement, random *rand.Rand, workerId string) {
 
+        msgPrinter := i18n.GetMessagePrinter()
+
+        // get node policy
+        nodePolicyHandler := exchange.GetHTTPNodePolicyHandler(b)
+        _, nodePolicy, err := compcheck.GetNodePolicy(nodePolicyHandler, wi.Device.Id, msgPrinter)
+
+        // If a deployment policy is being used, set wi.ProducerPolicy to the node policy
+        if wi.ConsumerPolicy.PatternId == "" {
+                // non pattern case
+
+                if err != nil {
+                        glog.Errorf(BAWlogstring(workerId, fmt.Sprintf("%v", err)))
+                        return
+                } else if nodePolicy == nil {
+                        glog.Warning(BAWlogstring(workerId, fmt.Sprintf("Cannot find node policy for this node %v.", wi.Device.Id)))
+                        return
+                } else {
+                        if glog.V(5) {glog.Infof(BAWlogstring(workerId, fmt.Sprintf("retrieved node policy: %v", nodePolicy)))}
```

Comment:
just use glog.V(5).Infof

Reply:
talked to Dave... its an optimization.. will leave as added